source/js/nav-newsletter.js

```diff
@@ -0,0 +1,135 @@
+import React from "react";
```

Comment:
This file feels off. Like it's trying to act like a prototyped object, without having been defined as such (using a `class`)

Reply:
Should I turn it into a class?